Util.py

```diff
@@ -2183,25 +2183,35 @@ def unicode_to_str(s):
   return s
 
 
-def deepcopy(x):
+def deepcopy(x, stop_types=None):
   """
   Simpler variant of copy.deepcopy().
   Should handle some edge cases as well, like copying module references.
 
   :param T x: an arbitrary object
+  :param list[type] stop_types: objects of these types will not be copied
   :rtype: T
   """
   # See also class Pickler from TaskSystem.
   # Or: https://mail.python.org/pipermail/python-ideas/2013-July/021959.html
   from TaskSystem import Pickler, Unpickler
 
+  persistent_memo = {}
+
+  def persistent_id(obj):
+    if type(obj) in stop_types:
```

Comment:
`stop_types` can also be `None`.
Also, maybe `isinstance` is better? It's not exactly the same, but I think `isinstance` would make more sense.

Reply:
If we also want to check for possible derived classes, yes. I am not sure if this exists.

Reply:
`isinstance` does exactly that. And usually it is recommend to use `isinstance` instead of `type(...) == ...` (PyCharm should also warn you about that). So I think it would follow that general convention (unless there are very good/explicit reasons to not do that).

Reply:
As the relation is "in", there is no warning, but I will change this to isinstance.